### PR TITLE
[Docs] add `initCap` examples 

### LINKED ISSUE #1965: Document `initcap` function.
Create documentation for the `initcap` function. See #1833 for details on how to complete this issue.

Child issue of #1833.

### PR DESCRIPTION
**Pull request by Blargian**
Closes [#1965](https://github.com/ClickHouse/clickhouse-docs/issues/1965) as part of [functions project](https://github.com/ClickHouse/clickhouse-docs/issues/1833)

Updates:
- `initCap`, `initCapUTF8`
- other minor changes to this page 
 
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/sql-reference/functions/string-functions.md [hunk 1]
```diff
@@ -571,16 +567,47 @@ SELECT upper('clickhouse');
...
-Does not detect the language, e.g. for Turkish the result might not be exactly correct (i/İ vs. i/I).
+:::note
+Does not detect the language, e.g. for Turkish the result might not be exactly correct (i/İ vs. i/I). If the length of the UTF-8 byte sequence is different for upper and lower case of a code point, the result may be incorrect for this code point.
+:::
 
-If the length of the UTF-8 byte sequence is different for upper and lower case of a code point, the result may be incorrect for this code point.
+**Syntax**
+
+```sql
+lowerUTF8(input)
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/functions/string-functions.md:577]: Should also contain a note about Turkish.

Also, add a note about letters represented by a different number of bytes in UTF-8, such as `ẞ` and `ß`.

**Comment by den-crane** [comment]
Also see https://github.com/ClickHouse/ClickHouse/issues/63419

File: docs/en/sql-reference/functions/string-functions.md [hunk 2]
```diff
@@ -12,9 +12,7 @@ Functions for [searching](string-search-functions.md) in strings and for [replac
 
 ## empty
 
-Checks whether the input string is empty.
-
-A string is considered non-empty if it contains at least one byte, even if this byte is a space or the null byte.
+Checks whether the input string is empty. A string is considered non-empty if it contains at least one byte, even if this byte is a space or the null byte.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/string-functions.md:15]: The form that works best with diff tools is

```md
Sentence 1 in paragraph 1.
Sentence 2 in paragraph 1.

Sentence 1 in paragraph 2.
Sentence 2 in paragraph 2.
```

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 03d6f37607b44a1c9bbbaae70a30d6077b913a15 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66166/03d6f37607b44a1c9bbbaae70a30d6077b913a15/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66166/03d6f37607b44a1c9bbbaae70a30d6077b913a15/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

**Pull request merged**